### ISSUE TITLE
Set `amp-story[publisher]` dynamically on the frontend

### ISSUE THREAD

**Issue by swissspidy** [GoogleForCreators/web-stories-wp#7190]
## Feature Description

In the editor, we set `amp-story[publisher]` to the site's name when generating the AMP markup.

That is fine, but means that whenever you update the site title in WordPress, this change won't be reflected on the frontend.

Even worse, things like schema.org metadata will be updated properly because it's generated dynamically, but `amp-story[publisher]` will always have the old value.

## Alternatives Considered

We could even update the editor to not use the site title, but a placeholder value like "Web Stories for WordPress", because it will always be replaced on the frontend. But there's no harm in keeping the current logic.

## Additional Context

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Implementation Brief

* Add new `\Google\Web_Stories\AMP\Traits\Sanitization_Utils::add_publisher()` method  
  Basically does  
  ```php
  $story_element = $document->body->getElementsByTagName( 'amp-story' )->item( 0 );
  $story_element->setAttribute( 'publisher-logo-src', get_bloginfo( 'name' ) );
  ```
* Call this new method in `Story_Sanitizer` and `AMP_Story_Sanitizer`
